src/brogue/Items.c

```diff
@@ -6841,6 +6956,10 @@ void readScroll(item *theItem) {
 }
 
 void detectMagicOnItem(item *theItem) {
+    if (theItem->category & (STAFF | WAND | POTION | SCROLL)) {
```

Comment:
This line makes me question the definition of HAS_FLAVORS. What does that mean? FIXED_POLARITY might make sense for these 4, but RING is problematic...

Reply:
I updated this to use HAS_FLAVORS, which means the base item is described with a random flavor so we don't know what it does. Rings are a bit nuanced, yeah. The way I think about it is the base ring type is always beneficial even if a particular ring instance is cursed. So for the purposes of identifying the base type, we can treat them the same as other flavored items.